controllers/etcd_controller.go

```diff
@@ -585,7 +590,7 @@ func (r *EtcdReconciler) reconcileStatefulSet(cm *corev1.ConfigMap, svc *corev1.
 			}
 		}
 
-		return ss.DeepCopy(), err
+		return r.waitUntilStatefulSetReady(ss)
```

Comment:
I still do no like this. But will park it for a separate PR later in the interest of getting druid out and unblocking gardener release.